tests/kernel/mem_protect/mem_protect/src/mem_domain.c

```diff
@@ -313,8 +313,7 @@ void mem_domain_for_user_tc3(void *max_partitions, void *p2, void *p3)
  */
 void test_mem_domain_add_partitions_invalid(void *p1, void *p2, void *p3)
 {
-
-	u8_t max_partitions = (u8_t)_arch_mem_domain_max_partitions_get();
+	u8_t max_partitions = (u8_t)_arch_mem_domain_max_partitions_get() - 1;
```

Comment:
Probably worth a comment here explaining the -1.  Or is there a way to get the number of partitions allocated via the API?  You could use that .    The commit message makes sense, but someone looking at only the code isn't going to understand the need for that offset or the fact that one domain is already initialized.